src/Microsoft.Health.Fhir.CosmosDb.UnitTests/Features/Storage/Registry/CosmosDbStatusRegistryInitializerTests.cs

```diff
@@ -0,0 +1,93 @@
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Microsoft.Health.Core;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+using Microsoft.Health.Fhir.CosmosDb.Features.Storage.Registry;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.CosmosDb.UnitTests.Features.Storage.Registry
+{
+    public class CosmosDbStatusRegistryInitializerTests
+    {
+        private CosmosDbStatusRegistryInitializer _initializer;
+        private ISearchParameterRegistry _searchParameterRegistry;
+        private ICosmosDocumentQueryFactory _cosmosDocumentQueryFactory;
+
+        public CosmosDbStatusRegistryInitializerTests()
+        {
+            _searchParameterRegistry = Substitute.For<ISearchParameterRegistry>();
+            _cosmosDocumentQueryFactory = Substitute.For<ICosmosDocumentQueryFactory>();
+
+            _initializer = new CosmosDbStatusRegistryInitializer(
+                () => _searchParameterRegistry,
+                _cosmosDocumentQueryFactory);
+
+            _searchParameterRegistry
+                .GetSearchParameterStatuses()
+                .Returns(new[]
+                {
+                    new ResourceSearchParameterStatus
+                    {
+                      Uri = new Uri("/test", UriKind.Relative),
+                      Status = SearchParameterStatus.Enabled,
+                      LastUpdated = Clock.UtcNow,
+                      IsPartiallySupported = false,
+                    },
+                });
+        }
+
+        [Fact]
+        public async Task GivenARegistryInitializer_WhenDatabaseIsExisting_NothingNeedsToBeDone()
+        {
+            IDocumentQuery<dynamic> documentQuery = Substitute.For<IDocumentQuery<dynamic>>();
+            _cosmosDocumentQueryFactory.Create<dynamic>(Arg.Any<IDocumentClient>(), Arg.Any<CosmosQueryContext>())
+                .Returns(documentQuery);
+
+            documentQuery
+                .ExecuteNextAsync()
+                .Returns(new FeedResponse<dynamic>(new dynamic[0]));
+
+            IDocumentClient documentClient = Substitute.For<IDocumentClient>();
+            var relativeCollectionUri = new Uri("/collection1", UriKind.Relative);
+
+            await _initializer.ExecuteAsync(documentClient, new DocumentCollection(), relativeCollectionUri);
+
+            await documentClient.Received().UpsertDocumentAsync(
+                relativeCollectionUri,
+                Arg.Any<SearchParameterStatusWrapper>());
+        }
+
+        [Fact]
+        public async Task GivenARegistryInitializer_WhenDatabaseIsNew_SearchParametersShouldBeUpserted()
```

Comment:
Is the title correct for this test case?  It seems like we are verifying that we did not upsert anything because the database was not new.